aesgcm does not compile on cygwin

```
$ make CFLAGS="-Werror -DUSE_LTM -DLTM_DESC -I../libtommath" EXTRALIBS=../libtommath/libtommath.a all
   * [...]
   * gcc demos/aesgcm.o
demos/aesgcm.c: In function ‘mv’:
demos/aesgcm.c:49:4: error: implicit declaration of function ‘syncfs’ [-Werror=implicit-function-declaration]
    syncfs(fd);
    ^~~~~~
cc1: all warnings being treated as errors
make: *** [makefile:49: demos/aesgcm.o] Error 1
```

I did not find a way how to use `syncfs()` from cygwin

--- a/demos/aesgcm.c
+++ b/demos/aesgcm.c
@@ -39,15 +39,27 @@ static off_t fsize(const char *filename)
    return -1;
 }
 
+#if defined(__linux__) && defined(__GLIBC_PREREQ)
+#if __GLIBC_PREREQ(2, 14)
+#define HAS_SYNCFS
+#endif
+#endif
+
 static int mv(const char *old_name, const char *new_name)
 {
    int fd;
    if (rename(old_name, new_name) == -1) return -1;
    fd = open(new_name, 0);
    if (fd == -1) return -1;
+#if !defined(_WIN32)
    if (fsync(fd) != 0) goto OUT;
+#if defined(HAS_SYNCFS)
    syncfs(fd);
+#else
+   sync();
+#endif
 OUT:
+#endif
    close(fd);
    return 0;
 }

